roles/ssl/tasks/main.yml

```diff
@@ -4,17 +4,17 @@
     path: /etc/nginx/ssl
     owner: root
     group: root
-    mode: '700'
+    mode: "700"
     state: directory
 
-- name: Generate 4096 bit DH parameters (this may take a long time...)
-  shell: openssl dhparam -writerand - 4096 > /etc/ssl/dhparam.pem
-  args:
-    chdir: /etc/nginx/ssl
-    creates: /etc/ssl/dhparam.pem
+- name: Download DH params
+  ansible.builtin.get_url:
+    url: https://ssl-config.mozilla.org/ffdhe2048.txt
```

Comment:
After doing some reading and chats looks like it is fine. I would suggest using the 4096bit one https://ssl-config.mozilla.org/ffdhe4096.txt

Reply:
I think this could be a parameter.

4096 has the disadvantage or larger bits is processing power needed. So it may make sense to use 2048 in some installs and 4096 in others.

Reply:
Made configurable in 87f8c49 with default to 2048 bits.